Patch by MSP-Greg
Allows test to find dlntest.dll when placed in install folder
diff --git a/test/-ext-/win32/test_dln.rb b/test/-ext-/win32/test_dln.rb
index e8f68ac4d4..301edf78ef 100644
--- a/test/-ext-/win32/test_dln.rb
+++ b/test/-ext-/win32/test_dln.rb
@@ -8,7 +8,9 @@ module Win32
     class TestDln < Test::Unit::TestCase
       def test_check_imported
         bug = '[Bug #6303]'
-        so = ::File.expand_path("../ext/-test-/win32/dln/dlntest.dll", ::EnvUtil.rubybin)
+        if !File.exist? (so = File.join(RbConfig::CONFIG['archdir'], '-test-/win32/dln/dlntest.dll'))
+          so = ::File.expand_path("../ext/-test-/win32/dln/dlntest.dll", ::EnvUtil.rubybin)
+        end
         assert_send([::File, :file?, so])
         path = ::ENV['PATH']
         path = ::File.dirname(so) + ::File::PATH_SEPARATOR + path
